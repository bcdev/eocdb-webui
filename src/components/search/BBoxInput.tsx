import { PureComponent } from "react";
import TextField from "@material-ui/core/TextField/TextField";
import * as React from "react";
import { Theme, withStyles, WithStyles } from "@material-ui/core";


import createStyles from "@material-ui/core/styles/createStyles";

export type BBoxValue = [number|undefined, number|undefined, number|undefined, number|undefined];


// noinspection JSUnusedLocalSymbols
const styles = (theme: Theme) =>  createStyles({
    searchField: {
        width: 300,
        marginLeft: theme.spacing.unit,
        marginRight: theme.spacing.unit,
    },
    textField: {},
    button: {},
    rightIcon: {},
    tableContainer: {},
});

interface BBoxInputProps extends WithStyles<typeof styles>{
    onBBoxChange: (selectedBBox: BBoxValue) => void;
    selectedBBox: BBoxValue;
}


class BBoxInput extends PureComponent<BBoxInputProps> {
    constructor(props: BBoxInputProps) {
        super(props);
    }


    handleSouthChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        let south = event.target.valueAsNumber;
        const {selectedBBox} = this.props;

        const newBBox: BBoxValue = [
            south,
            selectedBBox[1],
            selectedBBox[2],
            selectedBBox[3],
        ];

        this.props.onBBoxChange(newBBox);
    };

    handleWestChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const west = event.target.valueAsNumber;

        const {selectedBBox} = this.props;

        const newBBox: BBoxValue = [
            selectedBBox[0],
            west,
            selectedBBox[2],
            selectedBBox[3],
        ];
        this.props.onBBoxChange(newBBox);
    };

    handleNorthChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const north = event.target.valueAsNumber;
        const {selectedBBox} = this.props;

        const newBBox: BBoxValue = [
            selectedBBox[0],
            selectedBBox[1],
            north,
            selectedBBox[3],
        ];

        this.props.onBBoxChange(newBBox);
    };

    handleEastChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const east = event.target.valueAsNumber;
        const {selectedBBox} = this.props;

        const newBBox: BBoxValue = [
            selectedBBox[0],
            selectedBBox[1],
            selectedBBox[2],
            east,
        ];

        this.props.onBBoxChange(newBBox);
    };

    render() {
        const { classes, selectedBBox } = this.props;
        return (
            <div>
                <TextField
                    id={'bbox_bottom'}
                    label={'South'}
                    variant={'outlined'}
                    type={"number"}
                    className={classes.searchField}
                    onChange={this.handleSouthChange}
                    value={selectedBBox[0]}
                />
                <TextField
                    id={'bbox_left'}
                    label={'West'}
                    variant={'outlined'}
                    type={"number"}
                    className={classes.searchField}
                    onChange={this.handleWestChange}
                    value={selectedBBox[1]}
                />
                <TextField
                    id={'bbox_top'}
                    label={'North'}
                    variant={'outlined'}
                    type={"number"}
                    className={classes.searchField}
                    onChange={this.handleNorthChange}
                    value={selectedBBox[2]}
                />
                <TextField
                    id={'bbox_right'}
                    label={'East'}
                    variant={'outlined'}
                    type={"number"}
                    className={classes.searchField}
                    onChange={this.handleEastChange}
                    value={selectedBBox[3]}
                />
            </div>
        )
    }
}

export default withStyles(styles)(BBoxInput);