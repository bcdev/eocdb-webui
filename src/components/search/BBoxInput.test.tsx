import * as React from 'react';
import * as renderer from 'react-test-renderer';
import BBoxInput from "./BBoxInput";


interface BBoxInputProps{
    onBBoxChange: (left: number, bottom: number, right: number, top: number) => void;
    left: number;
    bottom: number;
    right: number;
    top: number;
}


describe('<BBoxInput />', () => {
    it('renders correctly', () => {
        let props: BBoxInputProps = {
            onBBoxChange: () => null,
            left: 0,
            bottom: 0,
            right: 0,
            top: 0,
        };

        const tree = renderer.create(
            <BBoxInput {...props} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
