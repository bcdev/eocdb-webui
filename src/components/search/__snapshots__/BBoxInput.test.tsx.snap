// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BBoxInput /> renders correctly 1`] = `
<div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-6 BBoxInput-searchField-1"
  >
    <label
      className="MuiFormLabel-root-21 MuiFormLabel-filled-25 MuiInputLabel-root-10 MuiInputLabel-formControl-15 MuiInputLabel-animated-18 MuiInputLabel-shrink-17 MuiInputLabel-outlined-20"
      data-shrink={true}
      htmlFor="bbox_left"
    >
      Left
    </label>
    <div
      className="MuiInputBase-root-41 MuiOutlinedInput-root-28 MuiInputBase-formControl-42"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="MuiPrivateNotchedOutline-root-58 MuiOutlinedInput-notchedOutline-35"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="MuiPrivateNotchedOutline-legend-59"
          style={
            Object {
              "width": 0,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInputBase-input-51 MuiOutlinedInput-input-36 MuiInputBase-inputType-54"
        defaultValue={undefined}
        disabled={false}
        id="bbox_left"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={false}
        rows={undefined}
        type="number"
        value={0}
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-6 BBoxInput-searchField-1"
  >
    <label
      className="MuiFormLabel-root-21 MuiFormLabel-filled-25 MuiInputLabel-root-10 MuiInputLabel-formControl-15 MuiInputLabel-animated-18 MuiInputLabel-shrink-17 MuiInputLabel-outlined-20"
      data-shrink={true}
      htmlFor="bbox_bottom"
    >
      Bottom
    </label>
    <div
      className="MuiInputBase-root-41 MuiOutlinedInput-root-28 MuiInputBase-formControl-42"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="MuiPrivateNotchedOutline-root-58 MuiOutlinedInput-notchedOutline-35"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="MuiPrivateNotchedOutline-legend-59"
          style={
            Object {
              "width": 0,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInputBase-input-51 MuiOutlinedInput-input-36 MuiInputBase-inputType-54"
        defaultValue={undefined}
        disabled={false}
        id="bbox_bottom"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={false}
        rows={undefined}
        type="number"
        value={0}
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-6 BBoxInput-searchField-1"
  >
    <label
      className="MuiFormLabel-root-21 MuiFormLabel-filled-25 MuiInputLabel-root-10 MuiInputLabel-formControl-15 MuiInputLabel-animated-18 MuiInputLabel-shrink-17 MuiInputLabel-outlined-20"
      data-shrink={true}
      htmlFor="bbox_right"
    >
      Right
    </label>
    <div
      className="MuiInputBase-root-41 MuiOutlinedInput-root-28 MuiInputBase-formControl-42"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="MuiPrivateNotchedOutline-root-58 MuiOutlinedInput-notchedOutline-35"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="MuiPrivateNotchedOutline-legend-59"
          style={
            Object {
              "width": 0,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInputBase-input-51 MuiOutlinedInput-input-36 MuiInputBase-inputType-54"
        defaultValue={undefined}
        disabled={false}
        id="bbox_right"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={false}
        rows={undefined}
        type="number"
        value={0}
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-6 BBoxInput-searchField-1"
  >
    <label
      className="MuiFormLabel-root-21 MuiFormLabel-filled-25 MuiInputLabel-root-10 MuiInputLabel-formControl-15 MuiInputLabel-animated-18 MuiInputLabel-shrink-17 MuiInputLabel-outlined-20"
      data-shrink={true}
      htmlFor="bbox_top"
    >
      Top
    </label>
    <div
      className="MuiInputBase-root-41 MuiOutlinedInput-root-28 MuiInputBase-formControl-42"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="MuiPrivateNotchedOutline-root-58 MuiOutlinedInput-notchedOutline-35"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="MuiPrivateNotchedOutline-legend-59"
          style={
            Object {
              "width": 0,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInputBase-input-51 MuiOutlinedInput-input-36 MuiInputBase-inputType-54"
        defaultValue={undefined}
        disabled={false}
        id="bbox_top"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={false}
        rows={undefined}
        type="number"
        value={0}
      />
    </div>
  </div>
</div>
`;
